File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.java
/*
    3 - 2
    1 - 1
    2 - 1
    총 4개 min(2개 키의 갯수 3개)
*/
import java.util.*;
class Solution {
    public int solution(int[] nums) {
        HashMap<Integer, Integer> map = new HashMap<>();
        for (int i : nums) {
            map.put(i, map.getOrDefault(i, 0) + 1);
        }
        
        return Math.min(map.size(), nums.length / 2);
    }
}

File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.cpp
#include <vector>
#include <unordered_set>
using namespace std;

int solution(vector<int> nums)
{
    unordered_set<int> s;
    for(int n : nums) {
        s.insert(n);
    }
    
    return min(s.size(), nums.size() / 2);
}